storage/src/desktop/storage_desktop.cc

```diff
@@ -134,6 +136,27 @@ void StorageInternal::CleanupCompletedOperations() {
   }
 }
 
+void StorageInternal::UseEmulator(const char* host, int port) {
+  if (host == nullptr || host[0] == '\0') {
+    throw std::invalid_argument("Emulator host cannot be null or empty.");
```

Comment:
We generally don't use C++ exceptions, with the philosophy that using them requires users to enable them on their end, which isn't guaranteed. I think the better thing to do would be to LogError, and return.